src/experiments/logins/api.js

```diff
@@ -0,0 +1,164 @@
+/**
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+
+/* globals AppConstants, browser, Components, CustomizableUI, dispatcher,
+   ExtensionCommon, ExtensionAPI, Services, XPCOMUtils */
+
+"use strict";
+
+ChromeUtils.defineModuleGetter(this, "ExtensionUtils",
+                               "resource://gre/modules/ExtensionUtils.jsm");
+ChromeUtils.defineModuleGetter(this, "LoginHelper",
+                               "resource://gre/modules/LoginHelper.jsm");
+ChromeUtils.defineModuleGetter(this, "Services",
+                               "resource://gre/modules/Services.jsm");
+
+const { ExtensionError } = ExtensionUtils;
+
+const getLogins = () => {
+  const logins = Services.logins.
+    getAllLogins().
+    filter((l) => !(l.hostname || "").startsWith("chrome://")).
+    map(LoginHelper.loginToVanillaObject);
+  return logins;
+};
+
+const getLogin = (id) => {
+  const login = getLogins().
+    filter(l => l.guid === id)[0];
+  return login || null;
+};
+
+this.logins = class extends ExtensionAPI {
+  getAPI(context) {
+    const EventManager = ExtensionCommon.EventManager;
+    return {
+      experiments: {
+        logins: {
```

Comment:
I realize this is not in the current doc ...

We have found it helpful to have a `touch(id)` method that updates the `timeLastUsed` (and `timesUsed`).  It's not hard to duplicate that logic further in the stack, but it's a common enough use that I think it's valuable to add here if we could.